controllers/components/kueue/kueue_controller.go

```diff
@@ -41,6 +44,7 @@ import (
 )
 
 func (s *componentHandler) NewComponentReconciler(ctx context.Context, mgr ctrl.Manager) error {
+	enableVAP = cluster.GetClusterInfo().Version.GTE(semver.MustParse("4.17.0"))
```

Comment:
since `enableVAP` is a package variable, it may be better to set it as part of the `init()` function